app/data_models/questionnaire_store.py

```diff
@@ -69,4 +74,11 @@ def delete(self):
 
     def save(self):
         data = self.serialize()
-        self._storage.save(data=data, submitted_at=self.submitted_at)
+        collection_exercise_sid = (
```

Comment:
This isn't the best. I would prefer to do this one when the state/store is created, but it's stuck in 

```
def get_questionnaire_store(user_id: str, user_ik: str) -> QuestionnaireStore:
    # Sets up a single QuestionnaireStore instance per request.
    store = g.get("_questionnaire_store")
    if store is None:
        secret_store = current_app.eq["secret_store"]  # type: ignore
        pepper = secret_store.get_secret_by_name(
            "EQ_SERVER_SIDE_STORAGE_ENCRYPTION_USER_PEPPER"
        )
        storage = EncryptedQuestionnaireStorage(user_id, user_ik, pepper)
        # pylint: disable=assigning-non-slot
        store = g._questionnaire_store = QuestionnaireStore(storage)

    return store
```
Unfortunately the metadata is not added in authenticator until after the state/store is created.  I could pass it into  get_questionnaire_store but we use that in lots of places just to get the store, effectively knowing it will be there, so passing in metadata isn't really appropriate. We perhaps need to consider separating out  the getting and the creating of state/store. Open to other ideas if anyone has them

Reply:
Given ce id isn't needed to identify the state, I think this is fine for where we are.